### PR TITLE
update discussion questions and resources

### PR DESCRIPTION
**Pull request by adamjonas**
based on winter/spring study group feedback

### CODE REVIEW

File: mining-network-prop/README.md [hunk 1]
```diff
@@ -12,24 +12,51 @@
...
+
+1. How can changes to Bitcoins p2p protocol affect mining?
+1. How is a high block-orphaning rate the best indicator that block propagation times are too high?
+1. Alternatively, while a low block-orphaning rate can be an indicator that block propagation times are good, it could also be an indicator of something more problematic. What could be a bad reason for there to be very few orphans?
+1. Greg Maxwell mentioned that miners are hesitant to connect with one another directly, even though it would speed up block propagation. Why?
+
+### Compact Blocks
+
+1. How could we speed up block propagation in the network like FIBRE does, but without the drawback of having to trust others?
+1. Compact block relay is a good way to reduce the amount of bandwidth to propagate new blocks to full nodes, but is it the fastest method for a synced node given peers can request missing transactions?
```
Comment (Xekyo) [inline mining-network-prop/README.md:39]: The "can request" is a bit confusing to me here. I assume it alludes to the recipient of a compact block potentially not having the full block. Then maybe:
 
```suggestion
1. Compact block relay is a good way to reduce the amount of bandwidth to propagate new blocks to full nodes, but is it the fastest method for a synced node given peers may need to request missing transactions?
```
would be better?

File: mining-network-prop/README.md [hunk 2]
```diff
@@ -12,24 +12,51 @@
...
 ## Discussion Questions
 
-1. How can changes to Bitcoins p2p protocol affect mining? (Advances in Block Propagation)
-1. How does the FIBRE network work and what were the motivations to iterate on the original Relay Network? (Relay Networks and FIBRE)
-1. How do pools give work assignments to client miners? Do all miners search the same block candidates?
-1. What are 'shares' and how do they help facilitate pooled mining?
-1. Consider the following scenario: Segwit2x didn’t happen (or miners didn’t throw in support for it) and UASF (BIP148) ended up causing a chain split on August 1st, 2017, with minority mining support (e.g. ~10-20%). What would likely happen as a result?
+### Mining and Consensus
+
+1. Why would some miners use parts of the version field as a nonce? What effect did this have on nodes?
```
Comment (Xekyo) [inline mining-network-prop/README.md:22]: Maybe that would fit better into the section on ASICBoost at the bottom?

File: p2p/README.md [hunk 3]
```diff
@@ -6,27 +6,32 @@
...
-[Deanonymization in the Bitcoin P2P Network](https://papers.nips.cc/paper/6735-deanonymization-in-the-bitcoin-p2p-network.pdf) | 15 min | P2P, privacy |
 [Eclipse Attacks on Bitcoin’s Peer-to-Peer Network](https://eprint.iacr.org/2015/263.pdf) | 60 min | eclipse attacks, P2P |
 [Network partitioning & network level privacy attacks](http://diyhpl.us/wiki/transcripts/chaincode-labs/2019-06-12-ethan-heilman-network-partitioning-attacks/) | 40 min | eclipse attacks, privacy, P2P |
 [Researching P2P privacy attacks](https://youtu.be/qKNEUfnYue0) | 90 min | P2P, privacy, video, _optional_ |
 
 ## Discussion Questions
 
+### Attacking P2P
+
+1. Given the attack surface in p2p, is running a full node worth it?
```
Comment (Xekyo) [inline p2p/README.md:21]:
```suggestion
1. Given the attack surface in P2P, is running a full node worth it?
```

File: p2p/README.md [hunk 4]
```diff
@@ -6,27 +6,32 @@
...
 1. What is the reasoning behind the max inbound and max outbound defaults? For which type of user would they be considered ideal, and when might they be optimized?
 1. What is the rationale behind the "new"/"tried" table design? Were there any prior inspirations within the field of distributed computing?
 1. How does a fixed set of 4 outbound peers get chosen? In what circumstances would you evict or change them?
 1. What are feeler connections, and when are they used?
 
-### De-anonymization
+### Eclipse Attacks on Bitcoin’s Peer-to-Peer Network
 
-1. How does "diffusion" message spreading work, and why is it ineffective against de-anonymization?
+1. What an attacker do if able to eclipse a mining pool?
```
Comment (Xekyo) [inline p2p/README.md:33]:
```suggestion
1. What can an attacker do if they are able to eclipse a mining pool?
```

File: script-wallets/README.md [hunk 5]
```diff
@@ -8,29 +8,38 @@
...
 1. Why is the internal chain not visible outside of the wallet if it uses public derivation?
 
 ### Coin Selection
+
 1. Is coin age ever a consideration for coin selection?
 1. Coin selection can expose a wallet by observing how the wallet selects its inputs, are there any efforts to standardize coin selection into a library of sorts so there's a standard?
+
+## Fee Bumping and RBF
+
+1. Is there way to ensure that a transaction will be processed? What tools are available ensure a stuck transaction (due to low fees) gets processed?
```
Comment (Xekyo) [inline script-wallets/README.md:44]:
```suggestion
1. Is there way to ensure that a transaction will be processed? What tools are available to ensure a stuck transaction (due to low fees) gets processed?
```

File: segwit/README.md [hunk 6]
```diff
@@ -13,24 +13,49 @@
 | [SegWit Costs](https://bitcoincore.org/en/2016/10/28/segwit-costs/) | 20 min | technical |
 | [The Long Road to SegWit](https://bitcoinmagazine.com/articles/long-road-segwit-how-bitcoins-biggest-protocol-upgrade-became-reality/) | 20 min | history |
 | [Bech32](https://diyhpl.us/wiki/transcripts/sf-bitcoin-meetup/2017-03-29-new-address-type-for-segwit-addresses/) | 35 min | bech32, *optional* |
-| [SegWit Wallet Development](https://bitcoincore.org/en/segwit_wallet_dev/) | 10 min | wallet, *optional* |
-| [Code Review: The Consensus Critical Parts of SegWit](https://petertodd.org/2016/segwit-consensus-critical-code-review) | 30 min | code, *optional* |
+| [SegWit Wallet Development](https://bitcoincore.org/en/segwit_wallet_dev/) | 10 min | wallet, _optional_ |
```
Comment (Xekyo) [inline segwit/README.md:16]: Is `*optional*` and `_optional_` equivalent?

File: segwit/README.md [hunk 7]
```diff
@@ -13,24 +13,49 @@
...
+### Transaction Malleability
+
+1. How is ECDSA malleable?
+
+### SegWit Benefits
+1. Why do second layer protocols (like Lightning) require transaction malleability to be fixed?
+
+### Advanced SegWit
+
+1. How is witness data committed to the block?
```
Comment (Xekyo) [inline segwit/README.md:39]:
```suggestion
1. How do blocks commit to witness data?
```

File: segwit/README.md [hunk 8]
```diff
@@ -13,24 +13,49 @@
...
+1. What was the quadratic sighash problem prior to Segwit? How does [BIP 143](https://github.com/bitcoin/bips/blob/master/bip-0143.mediawiki) solve this?
+1. What rationale was used to decide on the 4 MB SegWit block weight (3 x old_tx_bytes + segwit_tx_bytes), instead of say a 2 MB block weight (old_tx_bytes + segwit_tx_bytes)?
+
+### SegWit Costs
+
+1. How does SegWit affect initial block download (IBD)?
+
+### Gregory Maxwell’s Inhibition Proposal
+
+1. What is ASIC BOOST and what did it have to do with SegWit's deployment? ()
```
Comment (Xekyo) [inline segwit/README.md:53]:
```suggestion
1. What is ASIC BOOST and what did it have to do with SegWit's deployment?
```

File: segwit/README.md [hunk 9]
```diff
@@ -13,24 +13,49 @@
...
+
+### The Long Road to SegWit
+
+1. How did users know whether miners support SegWit prior to activation? ()
```
Comment (Xekyo) [inline segwit/README.md:57]:
```suggestion
1. How did users know whether miners support SegWit prior to activation?
```

File: welcome-to-the-bitcoin-protocol/README.md [hunk 10]
```diff
@@ -11,27 +11,45 @@
...
-1. What is Sybil Resistance and how does proof of work enable Sybil Resistant for new nodes joining the Bitcoin network?
+
+### Bitcoin's Security Model: A Deep Dive
+
+1. What do you think will happen when the mining reward runs out?
+
+### Security Models with John Newbery
+
+1. Define the differences between a full node, pruned node, and an SPV node?
+1. What are the incentives to run a full-node? If I do run a full-node, why accept incoming connections given that they come at a cost? What might those costs be?
```
Comment (Xekyo) [inline welcome-to-the-bitcoin-protocol/README.md:49]:
```suggestion
1. What are the incentives to run a full node? If I do run a full node, why accept incoming connections given that they come at a cost? What might those costs be?
```

File: mining-network-prop/README.md (hunk 2 above)
Comment (adamjonas) [reply mining-network-prop/README.md:22]: I'd like to keep this as is. As far as I understand, miners have used the version field as extra nonce space independent of ASICBoost.

**Pull request merged**